modules/components/analytics/core/org.wso2.carbon.pc.analytics.core/src/main/java/org/wso2/carbon/pc/analytics/core/TaskLevelMonitor.java

```diff
@@ -0,0 +1,483 @@
+						double avgExecTime = values.getDouble("avgExecutionTime");
+						table.put(taskDefKey, avgExecTime);
+					}
+					sortedResult = Helper.getDoubleValueSortedList(table, "taskDefId",
+					                                               "avgExecutionTime",
+					                                               order, taskCount);
+				}
+			}
+		} catch (Exception e) {
+			String errMsg = "PC Analytics core TaskLevelMonitoring error.";
+			log.error(errMsg, e);
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Result = " + sortedResult);
+		}
+		return sortedResult;
+	}
+
+	/**
+	 * perform query: SELECT taskDefinitionKey, COUNT(taskInstanceId) AS taskInstanceCount FROM
+	 *                USER_INVOLVE_SUMMARY_DATA GROUP BY taskDefinitionKey;
+	 * @param filters is used to filter the result
+	 * @return the result as a JSON string
+	 */
+	public String getTaskInstanceCountVsTaskId(String filters) {
+		String sortedResult = "";
+		try {
+			if (Helper.isDASAnalyticsActivated()) {
+				JSONObject filterObj = new JSONObject(filters);
+				String processId = filterObj.getString(AnalyticConstants.PROCESS_ID);
+				String order = filterObj.getString(AnalyticConstants.ORDER);
+				int taskCount = filterObj.getInt(AnalyticConstants.NUM_COUNT);
+
+				AggregateField countField = new AggregateField();
+				countField.setFieldName(AnalyticConstants.ALL);
+				countField.setAggregate(AnalyticConstants.COUNT);
+				countField.setAlias(AnalyticConstants.TASK_INSTANCE_COUNT);
+
+				ArrayList<AggregateField> aggregateFields = new ArrayList<>();
+				aggregateFields.add(countField);
+
+				AggregateQuery query = new AggregateQuery();
+				query.setTableName(AnalyticConstants.USER_INVOLVE_TABLE);
+				query.setGroupByField(AnalyticConstants.TASK_DEFINITION_KEY);
+				query.setQuery("processDefinitionId:" + "\"'" + processId + "'\"");
+				query.setAggregateFields(aggregateFields);
+
+				if (log.isDebugEnabled()) {
+					log.debug(Helper.getJSONString(query));
+				}
+
+				String result = AnalyticsRestClient
+								.post(Helper.getURL(AnalyticConstants.ANALYTICS_AGGREGATE),
+						        Helper.getJSONString(query));
+
+				JSONArray unsortedResultArray = new JSONArray(result);
+				Hashtable<String, Integer> table = new Hashtable<>();
+
+				if (unsortedResultArray.length() != 0) {
+					for (int i = 0; i < unsortedResultArray.length(); i++) {
+						JSONObject jsonObj = unsortedResultArray.getJSONObject(i);
+						JSONObject values = jsonObj.getJSONObject("values");
+						String processDefKey = values.getJSONArray("taskDefId").getString(0);
+						int processInstanceCount = values.getInt("taskInstanceCount");
+						table.put(processDefKey, processInstanceCount);
+					}
+					sortedResult = Helper.getIntegerValueSortedList(table, "taskDefId",
+					                                                "taskInstanceCount", order,
+					                                                taskCount);
+				}
+			}
+		} catch (Exception e) {
```

Comment:
Here from the post request (to the DAS rest API), I throw exeptions like UnsupportedEncodingException, UnsupportedOperationException.. to the outside. To capture the exeption from the Monitor class I used this generic Exception class.